codility_training/lessons.lesson07.StacksandQueues.Brackets/sangmandu.py

```diff
@@ -0,0 +1,25 @@
+# you can write to stdout for debugging purposes, e.g.
+# print("this is a debug message")
+
+def solution(S):
+    # write your code in Python 3.6
+    pass
+    st = []
+    for i in S:
+        if(len(st) == 0):
+            if(i == ")" or i == "]" or i == "}"):
+                return 0
+            st.append(i)
+        else:
+            if(i == ")"):
+                if(st.pop() != "("):
+                    return 0
+            elif(i == "}"):
+                if(st.pop() != "{"):
+                    return 0
+            elif(i == "]"):
+                if(st.pop() != "["):
+                    return 0
+            else:
+                st.append(i)

```

Comment:
```suggestion
        if(len(st) == 0):
            if(i == ")" or i == "]" or i == "}"):
                return 0
            st.append(i)
            continue;
        if i == ")" and st.pop() != "(" :
            return 0
        if i == "}" and st.pop() != "{":
            return 0
        if i == "]" and st.pop() != "[" :
            return 0
        st.append(i)
```
이렇게 if의 depth는 최대한 줄일 수 있도록 해주면 좋습니다!

추가로 dictionary를 이용한 방법도 있습니다.

```python
    dic = { "(": ")", "[": "]", "{": "}" }
    front = dic.keys();
```

이렇게 두 개의 변수만 있으면 if 문 한 번만 사용해도 된다는 점!

Reply:
이해 됐습니다!!
-dictionary도 사용가능!
- if문의 깊이를 줄일 것!
감사합니다